SimG4Core/Application/src/RunManagerMTWorker.cc

```diff
@@ -315,12 +310,11 @@ void RunManagerMTWorker::initializeG4(RunManagerMT* runManagerMaster, const edm:
 
   // setup the magnetic field
   if (m_pUseMagneticField) {
-    const GlobalPoint g(0., 0., 0.);
+    const GlobalPoint g(0.f, 0.f, 0.f);
 
-    edm::ESHandle<MagneticField> pMF;
-    es.get<IdealMagneticFieldRecord>().get(pMF);
+    const MagneticField* pMF = &(*es.getTransientHandle(m_MagField));
```

Comment:
To match what was done before, this one should just be `const MagneticField* pMF = &es.getData(m_MagField)` since it is likely the `MagneticField` object is not allowed to be deleted early.

Reply:
Calling this here can lead to failure as `initializeG4` can also be called from `OscarMTProducer::produce`

https://github.com/cms-sw/cmssw/blob/1e52da66e85bd9550c1a2b9afd801c4bbdf2d205/SimG4Core/Application/src/RunManagerMTWorker.cc#L506

The getting of the magentic field needs to be done in the begin run. This get should probably be done directly in OscarMTProducer and then passed to the RunManagerMTWorker.

Reply:
Done